File: inventory-frontend/src/services/notificationService.js
import api from './api';

const notificationService = {
  async getAllNotifications(page = 0, size = 10) {
    const response = await api.get('/api/notifications', {
      params: { page, size },
    });
    return response.data;
  },

  async getUnreadNotifications(page = 0, size = 10) {
    const response = await api.get('/api/notifications/unread', {
      params: { page, size },
    });
    return response.data;
  },

  async getUnreadCount() {
    const response = await api.get('/api/notifications/unread/count');
    return response.data;
  },

  async markAsRead(id) {
    const response = await api.put(`/api/notifications/${id}/read`);
    return response.data;
  },

  async markAllAsRead() {
    const response = await api.put('/api/notifications/read-all');
    return response.data;
  },
};

export default notificationService;

File: auth-module-initial-aws/microservice-notifications/src/main/java/com/example/microservice_notifications/service/NotificationService.java
package com.example.microservice_notifications.service;
import com.example.microservice_notifications.dto.NotificationDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface NotificationService {
    NotificationDto createNotification(NotificationDto dto);
    Page<NotificationDto> getAllNotifications(Pageable pageable);
    Page<NotificationDto> getUnreadNotifications(Pageable pageable);
    void markAsRead(UUID id);
    void markAllAsRead();
    Long getUnreadCount();
}